packages/plugins/users-permissions/server/controllers/auth.js

```diff
@@ -120,6 +120,63 @@ module.exports = {
     }
   },
 
+  async changePassword(ctx) {
+    const params = _.assign({}, ctx.request.body, ctx.params);
+
+    if (!_.get(ctx, 'state.user.id')) {
```

Comment:
the state is auto-defined you could be able to just do !ctx.state.user and return ctx.unauthorized here